require.lua

```diff
@@ -8,25 +8,25 @@ local t_concat = table.concat
 
 --- Helpers
 
-local function _tostring(s) return ""..s end
-
 local function checkstring(s)
-    local success, res = pcall(_tostring, s)
-    if success then return res end
-    error("bad argument #1 to 'require' (string expected, got "..type(s)..")", 3)
+    s = type(s) == "number" and tostring(s) or s -- support numerical module name
+    if type(s) ~= "string" then
+        error("bad argument #1 to 'require' (string expected, got "..type(s)..")", 3)
+    end
+    return s
 end
 
 --- for Lua 5.1
 
 local package, p_loaded = package, package.loaded
 
-local sentinel = newproxy and newproxy() or string.char(1,2,3,4,5)
+local sentinel = {} --newproxy and newproxy() or string.char(1,2,3,4,5)
 
 local function require51 (name)
-    name = checkstring(name)
```

Comment:
You can't get rid of this. `require(2)` looks for the string `"2"` in `package.loaded`.